sound/soc/sof/ipc4-topology.h

```diff
@@ -163,15 +163,16 @@ struct sof_ipc4_pin_format {
  * struct sof_ipc4_available_audio_format - Available audio formats
  * @out_audio_fmt: Available output audio format
  * @sink_audio_fmts: Available sink(input) audio formats
- * @ref_audio_fmt: Reference audio format to match runtime audio format
+ * @format_list_to_search: List of audio formats to match runtime params against. This points to
+ * either @out_audio_fmt or @input_audio_fmts
  * @audio_fmt_num: Number of available audio formats. Audio formats defined in topology always
  * contain both the input and output formats for a particular pin. So audio_fmt_num is the same
  * for both available input and output formats.
  */
 struct sof_ipc4_available_audio_format {
 	struct sof_ipc4_pin_format *out_audio_fmt;
 	struct sof_ipc4_pin_format *sink_audio_fmts;
-	struct sof_ipc4_pin_format *ref_audio_fmt;
+	struct sof_ipc4_pin_format *format_list_to_search;
```

Comment:
Now that you are cleaning up the definitions, I realized that we don't really use this "ref_audio_fmt" field elsewhere than in sof_ipc4_init_audio_fmt(). 

Could we just pass this as a parameter to sof_ipc4_init_audio_fmt() and not have it as part of sof_ipc4_available_audio_format struct...?